operations/system_configuration_service/Set_BMC_Credentials.md

```diff
@@ -1,97 +1,51 @@
-        xxx=`curl -k https://${fff}/redfish/v1/`
-        if [[ "${xxx}" != "" ]]; then
-            echo "PRESENT"
-        else
-            echo "NOT PRESENT"
-        fi
-    done
-    ```
-
-2.  Create a new JSON file containing the BMC credentials for all BMCs returned in the previous step.
-
-    Select one of the options below to set the credentials for the BMCs:
-
-    -   Set all BMCs with the same credentials.
-
-        ```bash
-        ncn-m001# vi bmc_creds_glb.json
-        {
-          "Force": false,
-          "Username": "root",
-          "Password": "new.root.password"
-          "Targets": [
-            "x0c0s0b0",
-            "x0c0s1b0"
-          ]
-        }
-        ```
-
-    -   Set all BMCs with different credentials.
-
-        ```bash
-        ncn-m001# vi bmc_creds_dsc.json
-        {
-          "Force": true,
-          "Targets": [
-            {
-              "Xname": "x0c0s0b0",
-              "Creds": {
-                "Username": "root",
-                "Password": "pw-x0c0s0b0"
-              }
-            },
-            {
-              "Xname": "x0c0s0b1",
-              "Creds": {
-                "Username": "root",
-                "Password": "pw-x0c0s0b1"
-              }
-            }
-          ]
-        }
-        ```
-
-3.  Apply the new BMC credentials.
-
-    Use only one of the following options depending on how the credentials are being set:
-
-    -   Apply global credentials to BMCs with the same credentials.
-
-        ```bash
-        ncn-m001# cray scsd bmc globalcreds create ./bmc_creds_glb.json
-        ```
-
-    -   Apply discrete credentials to BMCs with different credentials.
-
-        ```bash
-        ncn-m001# cray scsd bmc discreetcreds create ./bmc_creds_dsc.json
-        ```
-
-    **Troubleshooting:** If either command has any components that do not have the status of OK, they must be retried until they work, or the retries are exhausted and noted as failures. Failed modules need to be taken out of the system until they are fixed.
+- [Prerequisites](#prerequisites)
+- [Procedures](#procedures)
+  - [Generate a unique random password for each BMC in the system](#generate-a-unique-random-password-for-each-bmc-in-the-system)
+  - [Generate a single random password for all BMCs in the system](#generate-a-single-random-password-for-all-bmcs-in-the-system)
+  - [Provide a user-defined password for all BMCs in the system](#provide-a-user-defined-password-for-all-bmcs-in-the-system)
+
+## Prerequisites
+
+SAT is installed and configured.
+
+## Procedures
+
+Choose one of the following procedures.
+
+### Generate a unique random password for each BMC in the system
+
+```bash
+sat bmccreds --random-password --pw-domain bmc
+```
+
+### Generate a single random password for all BMCs in the system
+
+```bash
+sat bmccreds --random-password --pw-domain system
+```
+
+### Provide a user-defined password for all BMCs in the system
 
+```bash
+sat bmccreds USER-DEFINED_PASSWORD
```

Comment:
This needs a fix as well - to give the password on the command line, you need to use the `--password` argument, i.e.
```
sat bmccreds --password USER-DEFINED-PASSWORD
```

As it is written in the description of STP-2668, you run it with no positional arguments and the command prompts you to enter (and confirm) the new BMC password.

I could go either way on this but if we want to have the user enter the password on the command line it has to be with `--password`. I think it is better to have it prompt (run the command with no arguments) or at least to put the password in an environment variable, that way it does not necessarily appear in the command history.

And minor nit: The hyphen versus underscore in "USER-DEFINED_PASSWORD" if we keep it that way.

Reply:
I completely agree with you regarding not passing the password on the command-line.